tests/smoke-tests/test-relay-indices.ts

```diff
@@ -0,0 +1,226 @@
+      RELAY_ENCODER_CONTRACT_JSON.contract.abi
+    );
+
+    const XCM_TRANSACTOR_V1_JSON = getCompiled("XcmTransactorV1");
+    const XCM_TRANSACTOR_V1_INTERFACE = new ethers.utils.Interface(
+      XCM_TRANSACTOR_V1_JSON.contract.abi
+    );
+
+    const XCM_TRANSACTOR_V2_JSON = getCompiled("XcmTransactorV2");
+    const XCM_TRANSACTOR_V2_INTERFACE = new ethers.utils.Interface(
+      XCM_TRANSACTOR_V2_JSON.contract.abi
+    );
+
+    relayEncoder = new ethers.Contract(
+      RELAY_ENCODER_PRECOMPILE,
+      RELAY_ENCODER_INTERFACE,
+      context.ethers
+    );
+
+    xcmTransactorV1 = new ethers.Contract(
+      XCM_TRANSACTOR_V1_PRECOMPILE,
+      XCM_TRANSACTOR_V1_INTERFACE,
+      context.ethers
+    );
+
+    xcmTransactorV2 = new ethers.Contract(
+      XCM_TRANSACTOR_V2_PRECOMPILE,
+      XCM_TRANSACTOR_V2_INTERFACE,
+      context.ethers
+    );
+  });
+
+  it("should have matching indices for HRMP.InitOpenChannel", async function () {
+    if (rtVersion < 2100) {
+      debug(`Runtime version is ${rtVersion}, which is less than 2100. Skipping test. `);
+      this.skip();
+    }
+    const callIndex = context.relayApi.tx.hrmp.hrmpInitOpenChannel(2000, 1000, 102400).callIndex;
+    const callHex = u8aToHex(callIndex);
+    const resp = await relayEncoder.encodeHrmpInitOpenChannel(2000, 1000, 102400);
+    expect(resp.slice(0, 6), "Mismatched encoding between relaychain and local values").to.equals(
+      callHex
+    );
+  });
+
+  it("should have matching indices for HRMP.AcceptOpenChannel", async function () {
+    if (rtVersion < 2100) {
+      debug(`Runtime version is ${rtVersion}, which is less than 2100. Skipping test. `);
+      this.skip();
+    }
+    const callIndex = context.relayApi.tx.hrmp.hrmpAcceptOpenChannel(2001).callIndex;
+    const callHex = u8aToHex(callIndex);
+    const resp = await relayEncoder.encodeHrmpAcceptOpenChannel(2001);
+    expect(resp.slice(0, 6), "Mismatched encoding between relaychain and local values").to.equals(
+      callHex
+    );
+  });
+
+  it("should have matching indices for HRMP.CloseChannel", async function () {
+    if (rtVersion < 2100) {
+      debug(`Runtime version is ${rtVersion}, which is less than 2100. Skipping test. `);
+      this.skip();
+    }
+    const callIndex = context.relayApi.tx.hrmp.hrmpCloseChannel([2000, 2001]).callIndex;
+    const callHex = u8aToHex(callIndex);
+    const resp = await relayEncoder.encodeHrmpCloseChannel(2000, 2001);
+    expect(resp.slice(0, 6), "Mismatched encoding between relaychain and local values").to.equals(
+      callHex
+    );
+  });
+
+  it("should have matching indices for Staking.Bond", async function () {
+    const callIndex = context.relayApi.tx.staking.bond(ALITH_ADDRESS, 1000, "Staked").callIndex;
+    const callHex = u8aToHex(callIndex);
+    const resp = await relayEncoder.encodeBond(ALITH_ADDRESS, 1000, hexToU8a("0x00"));
+    expect(resp.slice(0, 6), "Mismatched encoding between relaychain and local values").to.equals(
```

Comment:
We should probably compare the whole response, that way we detect also mismatches between argument encoding too

Reply:
good idea, done